src/sysupdate/sysupdate-resource.c

```diff
@@ -183,13 +184,21 @@ static int resource_load_from_blockdev(Resource *rr) {
         if (r < 0)
                 return log_error_errno(r, "Failed to acquire partition table: %m");
 
+        /* Get disk sector size */
+        fd = fdisk_get_devfd(c);
+        if (fd < 0)
+                return log_error_errno(fd, "Failed to get device fd: %m");
+        r = blockdev_get_sector_size(fd, &sector_size);
+        if (r < 0)
+                return log_error_errno(r, "Failed to probe sector size: %m");
```

Comment:
So what you can do is use `fdisk_save_user_sector_size()` to save the sector size in the fdisk context and then query it with `fdisk_get_sector_size()`. Then you don't have to pass around the sector size separately.

Reply:
I didn't know `fdisk_get_sector_size()` was a thing. I believe just getting the value will be enough since the disk or image will already have a logical sector size set.

Reply:
Done.

Reply:
Getting just the value is not enough. We actually have a helper function fdisk_new_context_fd() or so in fdisk-util.h that should be used here.

Reply:
I meant getting just the sector size value with `fdisk_get_sector_size()`. Sorry for the confusion. This works as expected and passes the new test.